hibernate-reactive-core/src/main/java/org/hibernate/reactive/pool/impl/DefaultSqlClientPoolConfiguration.java

```diff
@@ -305,6 +295,45 @@ private String oraclePath(URI uri) {
 		return string.substring( start, end );
 	}
 
+	/**
+	 * if the host contains invalid characters {@link java.net.URI} parseAuthority(int start, int n) will end up
+	 * throwing a {@link java.net.URISyntaxException} which results in setting the host to NULL.
+	 * This method will reparse if host is NULL and set the host value
+	 */
+	private String findHost(URI uri, String scheme) {
+		if ( uri.getHost() == null ) {
+			String s = uri.toString();
+			String remainder = s.substring( s.indexOf( "://" ) + 3 );
+			// If no port defined than check for '/' or ';'
+			char endOfHostPortChar = ( scheme.equals( "sqlserver" )) ? ';' : '/';
+			String hostPortString = remainder.substring( 0, remainder.indexOf( endOfHostPortChar ) );
+			int endOfHost = hostPortString.indexOf( ':' );
+			if ( endOfHost == -1 ) {
+				return hostPortString;
+			}
+			return hostPortString.substring( 0, endOfHost );
+		}
+
+		return uri.getHost();
+	}
+
+	private int findPort(URI uri, String scheme) {
+		if ( uri.getPort() == -1 ) {
+			String s = uri.toString();
+			String remainder = s.substring( s.indexOf( "://" ) + 3 );
+			// If no port defined than check for '/' or ';'
+			char endOfHostPortChar = ( scheme.equals( "sqlserver" )) ? ';' : '/';
+			String hostPortString = remainder.substring( 0, remainder.indexOf( endOfHostPortChar ) );
+			int startOfPort = hostPortString.indexOf( ':' );
+			if ( startOfPort == -1 ) {
+				return -1;
+			}
+			return Integer.valueOf( hostPortString.substring( startOfPort+1 ) );
+		}
+
+		return uri.getPort();
```

Comment:
I don't understand all this extra logic. 
What's going on here?

Reply:
Sorry about it, I see that there was some extra logic even before.
I think we should collect all these cases in a  unique `findPort` method.
Something like:
```
int findPort(...) {
    int port = -1;
    if ("oracle".equals(scheme) ) {
        port = oraclePort(...);
    }
    else  if ("sqlServer".equals(scheme) ) {
        port = sqlServerPort(...);
    } else {
        port = ...
    }
    if ( port == -1) {
       return defaultPort();
    }
     return port;
}
```

It makes it easier to refactor and understand what's going on

Reply:
agreed